.github/workflows/publish.yml

```diff
@@ -14,40 +20,34 @@ jobs:
     - name: Set up Python
       uses: actions/setup-python@v1
       with:
-        python-version: 3.8
+        python-version: 3.x
 
     - name: Get tags
       run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
       shell: bash
 
     - name: Install build tools
       run: |
-        python -m pip install --upgrade pip wheel setuptools setuptools_scm twine "branca>=0.3.0" "jinja2>=2.9" numpy requests
+        python -m pip install --upgrade pip wheel build twine
 
       shell: bash
 
-    - name: Build binary wheel
-      run: pip wheel . -w dist --no-deps
-
-    - name: Build sdist
-      run: python setup.py sdist
+    - name: Build binary wheel and sdist
+      run: python -m build --sdist --wheel . --outdir dist
 
     - name: CheckFiles
       run: |
         ls dist
       shell: bash
 
-    - name: Check Version
-      run: |
-        python setup.py --version
-
     - name: Test wheels
       run: |
         cd dist && python -m pip install folium*.whl
         python -m twine check *
       shell: bash
 
     - name: Publish a Python distribution to PyPI
+      if: ${{ github.event_name == 'create' && github.event.ref_type == 'tag' }}
```

Comment:
I suppose this means when a tag is created? Seems like this would be better suited in the `on` part.

Reply:
Nope. We want to run all from above everywhere and this part below only on tags.

Reply:
I guess I'm not a fan of having this switching logic outside of the `on` statement, but it's okay 👍